lib/Dashboard.pm

```diff
@@ -68,11 +68,19 @@ sub startup ($self) {
   $self->renderer->compress(1);
 
   # Model
-  $self->helper(pg        => sub { state $pg        = Mojo::Pg->new($config->{pg})->max_connections(1) });
-  $self->helper(incidents => sub { state $incidents = Dashboard::Model::Incidents->new(pg => shift->pg) });
-  $self->helper(jobs      => sub { state $jobs     = Dashboard::Model::Jobs->new(pg => shift->pg, log => $self->log) });
-  $self->helper(settings  => sub { state $settings = Dashboard::Model::Settings->new(pg => shift->pg) });
-  $self->helper(amqp      => sub { state $amqp = Dashboard::Model::AMQP->new(log => $self->log, jobs => shift->jobs) });
+  $self->helper(pg => sub { state $pg = Mojo::Pg->new($config->{pg})->max_connections(1) });
+  $self->helper(incidents => sub ($c) { state $incidents = Dashboard::Model::Incidents->new(pg => $c->pg) });
+  $self->helper(
+    jobs => sub ($c) {
+      state $jobs = Dashboard::Model::Jobs->new(
+        days_to_keep_aggregates => $config->{days_to_keep_aggregates},
+        pg                      => $c->pg,
+        log                     => $self->log
+      );
+    }
+  );
+  $self->helper(settings => sub ($c) { state $settings = Dashboard::Model::Settings->new(pg => $c->pg) });
+  $self->helper(amqp => sub ($c) { state $amqp = Dashboard::Model::AMQP->new(log => $self->log, jobs => $c->jobs) });
```

Comment:
This only makes sense if `$c->jobs` is always the same object (regardless of `$c`). If that's the case you can ignore this comment.

(The same counts for `$c->pg` in previous lines but I suppose there it is really a given.)

Reply:
It's initialised with `state`, that means it is a constant value that doesn't change.

Reply:
I know. And that only makes sense if `$c->jobs` is independent from `$c` because otherwise the helper would always return the AMQP object for the first controller it has been invoked for.